src/metabase/driver.clj

```diff
@@ -149,6 +149,12 @@
      *  `:set-timezone`
      *  `:standard-deviation-aggregations`")
 
+  (field-avg-length ^Integer [this, ^FieldInstance field]
+    "*OPTIONAL*. Return the avergage length of FIELD. Default implementation does this in Clojure-land.")
+
+  (field-percent-urls ^Double [this, ^FieldInstance field]
```

Comment:
I'm not sure what reason you have for not wanting these in **the** interface. Passing them as arguments doesn't let us do anything we couldn't otherwise do and essentially ends up creating an undocumented ad-hoc informal interface.

It makes it easier to catch bugs too. It turns out the druid driver wasn't even running this part of the sync process, so I was able to fix some bugs there too.